source/opt/aggressive_dead_code_elim_pass.h

```diff
@@ -44,7 +44,12 @@ class AggressiveDCEPass : public MemPass {
   using GetBlocksFunction =
       std::function<std::vector<BasicBlock*>*(const BasicBlock*)>;
 
-  AggressiveDCEPass();
+  // preserve_bindings and preserve_spec_constants here serve as overrides
+  // for context()->preserve_bindings() and context->preserve_spec_constants(),
+  // since e.g. when legalizing HLSL we never want to strip bindings or
+  // specialization constants.
```

Comment:
When we did all of the work for legalization, we had the specific requirement that we were to remove unused resources.  It takes time to bind resources, so if the resource is unused, you can save time by not binding the resource.

Correct me if I am wrong, but removing the binding was also the default behaviour for FXC and DXC when generating DXIL.  That should be the default behaviour.